trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/result/ExpressionResultTransformer.java

```diff
@@ -1147,4 +1076,104 @@ ExpressionResult apply(final ExpressionResultTransformer ert, final ExpressionRe
 				final CType targetCType, final ILocation loc, final IASTNode hook);
 	}
 
+	public interface IPointerReadWrite {
+		ExpressionResult getDispatchedResult();
+
+		ExpressionResult getRead();
+
+		ExpressionResult getWrite(final Function<CType, ExpressionResult> valueProvider);
+
+		default ExpressionResult getWrite(final Expression value) {
+			return getWrite(type -> new ExpressionResult(new RValue(value, type)));
+		}
+	}
+
+	private class NormalPointerReadWrite implements IPointerReadWrite {
```

Comment:
Can we rename this? I think it is not clear what "normal" means here (at least, I don't know). Also, even after renaming, please document the class.